src/server.c

```diff
@@ -3252,10 +3231,65 @@ void InitServerLast() {
     server.initial_memory_usage = zmalloc_used_memory();
 }
 
+/* The purpose of this function is to try to "glue" consecutive range
+ * key specs in order to build the legacy (first,last,step) spec
+ * used by the COMMAND command.
+ * By far the most common case is just one range spec (e.g. SET)
+ * but some commands' ranges were split into two or more ranges
+ * in order to have different flags for different keys (e.g. SMOVE,
+ * first key is "read write", second key is "write").
+ *
+ * This functions uses very basic heuristics and is "best effort":
```

Comment:
the best effort here is to come up with a legacy range that is as broad as possible.
the only use for that legacy range is the backwards compatible response of the COMMAND command.
i.e. clients that look only at indexes 4,5,6 need to still be able to find there what they're looking for.
so instead of maintaining these legacy ranges in the command table, we generate them at runtime by looking for consecutive ranges.

That comment was meant to tell that story, so if it wasn't clear enough, maybe you can suggest a phrase that will fill in the gap that caused the confusion?